src/bin/elasticsearch_exporter.rs

```diff
@@ -0,0 +1,359 @@
+
+impl FromStr for HashMapDuration {
+    type Err = SimpleError;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        let mut map = ExporterPollIntervals::new();
+
+        let parts = input.trim().split("&").into_iter().collect::<Vec<&str>>();
+
+        for part in parts.into_iter() {
+            match part.split_once("=") {
+                Some((key, value)) => match value.parse::<humantime::Duration>() {
+                    Ok(time) => {
+                        let _ = map.insert(key.to_string(), *time);
+                    }
+                    Err(e) => {
+                        return Err(SimpleError(format!(
+                            "Failed to parse time for key {} value {} err {}",
+                            key, value, e
+                        )))
+                    }
+                },
+                None => {
+                    return Err(SimpleError(format!(
+                        "Usage `{}`, you provided `{}`",
+                        HASH_MAP_STR_FORMAT, part
+                    )))
+                }
+            }
+        }
+
+        Ok(Self(map))
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+struct HashMapVec(CollectionLabels);
+
+impl FromStr for HashMapVec {
+    type Err = SimpleError;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        let mut map = CollectionLabels::new();
+
+        let parts = input.trim().split("&").into_iter().collect::<Vec<&str>>();
+
+        for part in parts.into_iter() {
+            match part.split_once("=") {
+                Some((key, value)) => {
+                    let labels = value
+                        .split(",")
+                        .map(|value| value.to_string())
+                        .collect::<Vec<String>>();
+
+                    if let Some(values) = map.get_mut(key) {
+                        values.extend(labels);
+                    } else {
+                        let _ = map.insert(key.to_string(), labels);
+                    }
+                }
+                None => {
+                    return Err(SimpleError(format!(
+                        "Usage `{}`, you provided `{}`",
+                        HASH_MAP_STR_FORMAT, part
+                    )))
+                }
+            }
+        }
+
+        Ok(Self(map))
+    }
+}
+
+#[derive(Clone, Debug, Default)]
+struct HashMapStr(Labels);
+
+#[derive(Debug)]
+pub struct SimpleError(String);
+
+impl fmt::Display for SimpleError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl StdError for SimpleError {}
+
+const HASH_MAP_STR_FORMAT: &'static str = "cat_indices=id,pri,rep&cat_nodes=heap.percent,jdk";
+
+impl FromStr for HashMapStr {
+    type Err = SimpleError;
+
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        let mut map = Labels::new();
+
+        let parts = input.trim().split("&").into_iter().collect::<Vec<&str>>();
```

Comment:
I'd say instead of splitting this manually you could opt for https://docs.rs/serde_qs/0.7.0/serde_qs/, I believe it should work.

Reply:
@buinauskas have tried serde_eq works great for simple types, for Vec and "Human" duration had use custom, because for Vec query string expects to receive in format `&user_ids[0]=1&user_ids[1]=2` which is tedious to write to.